lib/rubocop/cop/style/disable_cop_for_entire_file_in_source_code.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    # rubocop:disable Lint/RedundantCopDisableDirective
+    module Style
+      # Enforces that disabling cops for an entire file happens in a config
+      # file, rather than by wrapping the entire file in `rubocop:disable` and
+      # `rubocop:enable` comments.
+      #
+      # @example
+      #   # bad
+      #   # rubocop:disable Department/CopName
+      #   entire_file
+      #   # rubocop:enable Department/CopName
+
+      #   # bad
+      #   # rubocop:todo Department/CopName
+      #   entire_file
+      #   # rubocop:enable Department/CopName
+      #
+      #   # good
+      #   # Department/CopName disabled in config, such as .rubocop.yml or .rubocop_todo.yml
+      #   entire_file
+      #
+      #   # good
+      #   code
+      #   # rubocop:disable Department/CopName
+      #   subset_of_code
+      #   # rubocop:enable Department/CopName
+      #   more_code
+      class DisableCopForEntireFileInSourceCode < Base
+        # rubocop:enable Lint/RedundantCopDisableDirective
+        MSG = 'When disabling cops for an entire file, prefer excluding the file in `.rubocop.yml`.'
+        TODO_MSG = 'When marking cops as TODO for an entire file, ' \
+                   'prefer excluding the file in `.rubocop_todo.yml`.'
+
+        def on_new_investigation
+          return if disabled_ranges.empty?
+
+          disabled_ranges.each do |disabled_range|
+            next unless disabled_range.cover?(code_range)
+
+            on_offending_comment(range_disabling_comment(disabled_range))
+          end
+        end
+
+        private
+
+        def on_offending_comment(comment)
+          return add_offense(comment, message: TODO_MSG) if comment.text.include?('rubocop:todo')
```

Comment:
Not sure if there's some constant I should refer to, or helper method, rather than hard code it. Perhaps something in `CommentConfig`? 🤔

Reply:
You could probably add a `todo?` method to `DirectiveComment` and then call `DirectiveComment.new(comment).todo?`

Reply:
I implemented this, but later switched to your suggested message, which doesn't require specialization. However, I'll see if it turns out to be useful anyway, and add it in another PR.